tests/backup/backup_test.go

```diff
@@ -5403,19 +5405,20 @@ var _ = Describe("{ReplicaChangeWhileRestore}", func() {
 			ctx, err := backup.GetAdminCtxFromSecret()
 			log.FailOnError(err, "Fetching px-central-admin ctx")
 			for _, namespace := range bkpNamespaces {
-				backupName = fmt.Sprintf("%s-%s-%v", BackupNamePrefix, namespace, time.Now().Unix())
-				restoreName = fmt.Sprintf("%s-%s-%v", restoreNamePrefix, backupName, time.Now().Unix())
-				scName := fmt.Sprintf("replica-sc-%v", time.Now().Unix())
-				pvcs, err := core.Instance().GetPersistentVolumeClaims(namespace, labelSelectors)
-				singlePvc := pvcs.Items[0]
-				dash.VerifyFatal(err, nil, "Getting PVC from namespace")
-				sourceScName, err := core.Instance().GetStorageClassForPVC(&singlePvc)
-				dash.VerifyFatal(err, nil, "Getting SC from PVC")
-				storageClassMapping[sourceScName.Name] = scName
-				restoredNameSpace := fmt.Sprintf("%s-%s", namespace, "restored")
-				namespaceMapping[namespace] = restoredNameSpace
-				err = CreateRestore(restoreName, backupName, namespaceMapping, SourceClusterName, orgID, ctx, storageClassMapping)
-				dash.VerifyFatal(err, nil, "Restoring with custom Storage Class Mapping")
+				for _, backupName := range backupNames {
+					restoreName = fmt.Sprintf("%s-%s-%v", restoreNamePrefix, backupName, time.Now().Unix())
```

Comment:
Can we move time.Now().Unix() call and assign it to a variable? Here it'll keep changing with every call.

Reply:
We tried that. The problem with declaring it at the top of Describe is that ginkgo initialised all the variables at describe level at the same time which will cause a lot of issues and create backups with duplicate names. We tried it and it started failing.
Here we are creating backup with guaranteed unique names and store it in a list and iterate over it.